exercises/leap/canonical-data.json

```diff
@@ -1,39 +1,29 @@
 {
    "cases": [
       {
-         "description": "leap year in twentieth century",
-         "input": 1996,
-         "expected": true
-      },
-      {
-         "description": "odd standard year in twentieth century",
-         "input": 1997,
-         "expected": false
-      },
-      {
-         "description": "even standard year in twentieth century",
-         "input": 1998,
+         "description": "year not divisible by 4: standard year",
+         "input": 2015,
          "expected": false
       },
       {
-         "description": "standard year in nineteenth century",
-         "input": 1900,
-         "expected": false
+         "description": "year divisible by 4, not divisible by 100: leap year",
```

Comment:
I still disagree - we have a decision tree, where the **next** rule ("unless") leads to an exemption of the current rule.

So 
-  rule 1 fails => common year: only one rule to check, so *only one rule in the description*
- rule 1 succeeds => possibly a leap year, we need to check rule 2
  - rule 2 fails: no exemption, so the result of rule 1 holds => leap year, no need to check further roles, so *rule 1 and rule 2 in the description* 
  - rule 2 succeeds => exemption, possibly a common year, we need to check rule 3
    - rule 3 fails: no exemption, the result of rule 2 holds => common year, I mention the *two really relevant rules* (divisibility by 4 is obvious from mathematics if the number is divisible by 100)
    - rule 3 succeeds: exemption =>leap year *only the last rule is really relevant* (divisibility by 4 and by 100 is (again) obvious from mathematics)

:+1 I really appreciate how you force me to rethink the issue.